src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/PublishItemsOutputGroupProvider.cs

```diff
@@ -9,72 +9,45 @@ namespace Microsoft.VisualStudio.ProjectSystem.Build
     [Order(Order.Default)]
     internal class PublishItemsOutputGroupProvider : IOutputGroupProvider
     {
+        private const string PublishItemsOutputGroupTargetName = "PublishItemsOutputGroup";
+
         /// <summary>
-        /// List of well known output groups names and their associated target and description
+        /// Collection containing a single "publish items" output group.
+        /// This is a singleton instance, shared across all projects/configurations.
         /// </summary>
-        private static readonly ImmutableHashSet<IOutputGroup> s_outputGroups = ImmutableHashSet.Create<IOutputGroup>()
-            .Add(NewGroup("PublishItems", "PublishItemsOutputGroup", Resources.OutputGroupPublishItemsDisplayName, Resources.OutputGroupPublishItemsDescription));
+        private static readonly ImmutableHashSet<IOutputGroup> s_outputGroups = ImmutableHashSet.Create<IOutputGroup>(
+            new OutputGroup(
+                name: "PublishItems",
+                targetName: PublishItemsOutputGroupTargetName,
+                displayName: Resources.OutputGroupPublishItemsDisplayName,
+                description: Resources.OutputGroupPublishItemsDescription,
+                items: ImmutableList<KeyValuePair<string, IImmutableDictionary<string, string>>>.Empty,
+                successful: false));
 
         private readonly AsyncLazy<IImmutableSet<IOutputGroup>> _outputGroups;
-        private readonly IProjectAccessor _projectAccessor;
-        private readonly ConfiguredProject _configuredProject;
 
         [ImportingConstructor]
-        internal PublishItemsOutputGroupProvider(IProjectAccessor projectAccessor, ConfiguredProject configuredProject, IProjectThreadingService projectThreadingService)
-        {
-            _projectAccessor = projectAccessor;
-            _configuredProject = configuredProject;
-            _outputGroups = new AsyncLazy<IImmutableSet<IOutputGroup>>(GetOutputGroupMetadataAsync, projectThreadingService.JoinableTaskFactory);
-        }
-
-        public Task<IImmutableSet<IOutputGroup>> OutputGroups
-        {
-            get { return _outputGroups.GetValueAsync(); }
-        }
-
-        /// <summary>
-        /// Gets a collection of names of targets in this project.
-        /// </summary>
-        /// <returns>Collection of the names of targets in this project.</returns>
-        private Task<ImmutableHashSet<string>> GetProjectTargetsAsync()
-        {
-            return _projectAccessor.OpenProjectForReadAsync(_configuredProject, project =>
-                project.Targets.Keys.ToImmutableHashSet(StringComparers.TargetNames));
-        }
```

Comment:
This is where we used to allocate a 500 element `ImmutableHashSet<string>`, which no longer occurs.